src/main/java/uk/gov/justice/probation/courtcasematcher/model/domain/Address.java

```diff
@@ -12,11 +12,89 @@
 @NoArgsConstructor(access = AccessLevel.PRIVATE, force = true)
 @Builder
 @JsonInclude(JsonInclude.Include.NON_NULL)
-public class Address {
+public class Address implements Comparable<Address> {
     private final String line1;
     private final String line2;
     private final String line3;
     private final String line4;
     private final String line5;
     private final String postcode;
+
+    @Override
+    public int compareTo(Address addressToCompare) {
+        if (this.getLine1() == null && addressToCompare.getLine1() == null) {
```

Comment:
This has produced a big old chunk of code given we're only really interested in equality of a subset of fields 🤔 In retrospect I wonder whether Comparable was a bit of a heavyweight tool for what we needed here but never mind, it will do the job!

Reply:
I had these checks inside the Defender comparator initially, but I couldn't find a way to check the null address first before accessing its properties. yes, I dislike comparable, but this makes the main comparator neet.

` .thenComparing(d -> d.getAddress().getLine1(), nullsFirst(naturalOrder()))
            .thenComparing(d -> d.getAddress().getLine2(), nullsFirst(naturalOrder()))
            .thenComparing(d -> d.getAddress().getLine3(), nullsFirst(naturalOrder()))
            .thenComparing(d -> d.getAddress().getLine4(), nullsFirst(naturalOrder()))
            .thenComparing(d -> d.getAddress().getLine5(), nullsFirst(naturalOrder()))
            .thenComparing(d -> d.getAddress().getPostcode(), nullsFirst(naturalOrder())));
`